service/api_task.go

```diff
@@ -654,68 +677,78 @@ func handleOldAgentRevision(response apimodels.NextTaskResponse, h *host.Host, w
 					"revision":  evergreen.BuildRevision,
 				}))
 				gimlet.WriteResponse(w, gimlet.MakeJSONInternalErrorResponder(innerErr))
-				return apimodels.NextTaskResponse{}, true
+				return nil, true
```

Comment:
Did you change these to nil because this has some effect on the caller, or to simplify?

Reply:
This  is a misleading diff--this function is getDetails, which is returning `*apimodels.GetNextTaskDetails` rather than `NextTaskResponse` (which is handled in `handleOldAgentRevision`) which is why I return nil. `handleOldAgentRevision` still returns `apimodels.NextTaskResponse{}`.